syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter 
{
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  
  rpc AcceptInt (IntRequest) returns (CustomResponse) {}

  rpc AcceptDouble (DoubleRequest) returns (CustomResponse) {}

  rpc AcceptString (StringRequest) returns (CustomResponse) {}

  rpc AcceptComplexDataStructure (ComplexDataStructureRequest) returns (CustomResponse) {}

  rpc AcceptClientSideStream (stream StringRequest) returns (CustomResponse) {}
}

// The request message containing the user's name.
message HelloRequest 
{
  string name = 1;
}

// The response message containing the greetings
message HelloReply 
{
  string message = 1;
}

message IntRequest
{
  int32 val = 1;
}

message DoubleRequest
{
  double val = 1;
}

message StringRequest
{
  string val = 1;
}

message ComplexDataStructureRequest
{
  int32 val1 = 1;
  double val2 = 2;
  string val3 = 3;
}

message CustomResponse
{
  string val = 1;
}
